Source/Language/Flow/SetupSequencePhrase.cs

```diff
@@ -93,6 +95,38 @@ public ISetupSequentialResult<TResult> Returns(TResult value)
 			this.setup.AddReturns(value);
 			return this;
 		}
+		public ISetupSequentialResult<TResult> Returns(Delegate valueFunction)
+		{
+			// If `TResult` is `Delegate`, that is someone is setting up the return value of a method
+			// that returns a `Delegate`, then we have arrived here because C# picked the wrong overload:
+			// We don't want to invoke the passed delegate to get a return value; the passed delegate
+			// already is the return value.
+			if (typeof(TResult) == typeof(Delegate))
+			{
+				return this.Returns(() => (TResult)(object)valueFunction);
+			}
+
+			// The following may seem overly cautious, but we don't throw an `ArgumentNullException`
+			// here because Moq has been very forgiving with incorrect `Returns` in the past.
+			if (valueFunction == null)
+			{
+				return this.Returns(() => default(TResult));
+			}
+
+			if (valueFunction.GetMethodInfo().ReturnType == typeof(void))
```

Comment:
If the parameter type is changed to `Func<TResult>`, then this `if` will become redundant.